api/v3/utils.php

```diff
@@ -461,6 +461,382 @@ function _civicrm_api3_store_values(&$fields, &$params, &$values) {
 /**
  * Get function for query object api.
  *
+ * This is a simple get function, but it should be usable for any kind of
+ * entity. I created it to work around CRM-16036.
+ *
+ * @param string $dao_name
+ *   Name of DAO
+ * @param array $params
+ *   As passed into api get function.
+ * @param bool $isFillUniqueFields
+ *   Do we need to ensure unique fields continue to be populated for this api? (backward compatibility).
+ * @param array $extraMysql
```

Comment:
Good point. Do you want to PR some of this stuff in as a code comment because I'm only going to work on it again now if it is either broken or insecure (due to needing to prioritise other things) - but I don't want to lose these recommendations.